hype-submitter/src/main/java/com/spotify/hype/runner/KubernetesDockerRunner.java

```diff
@@ -256,38 +257,56 @@ private Pod getBasePod(RunEnvironment env, String image) {
     }
   }
 
-  private VolumeMountInfo volumeMountInfo(PersistentVolumeClaim claim, VolumeMount volumeMount) {
-    final String claimName = claim.getMetadata().getName();
-
-    final Volume volume = new VolumeBuilder()
-        .withName(claimName)
-        .withNewPersistentVolumeClaim(claimName, volumeMount.readOnly())
-        .build();
+  private VolumeMountInfo volumeMountInfo(Volume volume, VolumeMount volumeMount) {
 
     final io.fabric8.kubernetes.api.model.VolumeMount mount = new VolumeMountBuilder()
-        .withName(claimName)
+        .withName(volume.getName())
         .withMountPath(volumeMount.mountPath())
         .withReadOnly(volumeMount.readOnly())
         .build();
 
     final String ro = volumeMount.readOnly() ? "readOnly" : "readWrite";
-    LOG.info("Mounting {} {} at {}", claimName, ro, volumeMount.mountPath());
+    LOG.info("Mounting {} {} at {}", volume.getName(), ro, volumeMount.mountPath());
 
     return new VolumeMountInfoBuilder()
-        .persistentVolumeClaim(claim)
         .volume(volume)
         .volumeMount(mount)
         .build();
   }
 
-  private List<VolumeMountInfo> volumeMountInfos(List<VolumeMount> volumeMounts) {
-    final Map<VolumeRequest, PersistentVolumeClaim> claims = volumeMounts.stream()
-        .map(VolumeMount::volumeRequest)
-        .distinct()
-        .collect(toMap(identity(), volumeRepository::getClaim));
+  private Volume buildVolume(VolumeMount volumeMount) {
+    final com.spotify.hype.model.Volume volume = volumeMount.volume();
+
+    if (volume instanceof VolumeRequest) {
+      final VolumeRequest volumeRequest = (VolumeRequest) volume;
+      final PersistentVolumeClaim claim = volumeRepository.getClaim(volumeRequest);
+      final String claimName = claim.getMetadata().getName();
+      return new VolumeBuilder()
+          .withName(claimName)
+          .withNewPersistentVolumeClaim(claimName, volumeMount.readOnly())
+          .build();
+    } else if (volume instanceof PersistentDisk) {
```

Comment:
Very interesting point, this was my original idea but it turned out that handling the K8s volume from the Volume interface turned our to be hairy because the VolumeRequest relies on a VolumeRepository that's not exposed to the user. 
Probably a way around it, let's discuss it.